android/configure_java.md

```diff
@@ -14,7 +14,7 @@ Caveat: Robolectric warns that Java 9 is needed for certain functionality. This
 ### Method #1: configure Java 8 from Android Studio
 Do the following:
 * Download and install Android Studio
-* Add the following line to your `~/.zshrc` or equivalent shell startup file: `export JAVA_HOME="/Applications/Android Studio.app/Contents/jbr/Contents/Home"`
+* Add the following line to your `~/.zshrc` or equivalent shell startup file: `export JAVA_HOME="/Applications/Android Studio.app/Contents/**/Contents/Home"`
```

Comment:
this sounds better imo. One more little improvement would be to mention that "**" needs to be replaced accordingly. A copy-paste might end up with "**" :)

Reply:
Double-astericks in most shells means that it should look for the `/Contents/Home` in any of the directories or sub-directories. This means you can copy-paste it into your shell safely. 🙂

Reply:
Drive-by: I'm not a shell wizard 🧙‍♀️ so I might be wrong but I think `**` is a shell expansion, i.e. if two paths are matched, this would expand to include multiple paths, for example something like `JAVA_HOME="/Applications/Android Studio.app/Contents/macOS/Contents/Home /Applications/Android Studio.app/Contents/Resources/Contents/Home"`, and I assume that'd cause the JAVA_HOME env var to not work.

But this is only possible if Android Studio is the type of package that would ship two paths that match `**/Contents/Home` (two JDKs?) and I don't know if it is. 🤷

Reply:
Fair point @mcomella ! I was considering whether to use `*` which is for current directories or `**` which is for current directory and the sub-directories to limit that kind of confusion.

I would say it might be safe to assume it would only ship with one. The only situation I can think of where there could be multiple JDKs would be to support multi-arch devices and this is something that is packaged _internally to the java library_.

When setting up appservices the other day, I learnt about this:
```
$ which java | xargs file

 /usr/bin/java: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64
 - Mach-O 64-bit executable x86_64] [arm64e:Mach-O 64-bit executable arm64e
 - Mach-O 64-bit executable arm64e]
 /usr/bin/java (for architecture x86_64):        Mach-O 64-bit executable x86_64
 /usr/bin/java (for architecture arm64e):        Mach-O 64-bit executable arm64e
```

Is there another situation we should consider? Any preference towards `*` or `**`?

Reply:
Ah, I was not aware of double-asterisk usage in the shell commands, thanks for the info. However, with this path, I was not able to make java executable working on my local machine.  As a result, I still needed to specify the exact path by replacing double-asterisk with `jre`. 
Regarding to the issue @mcomella mentioned, it is not the default case for now as Android Studio is shipped with one jdk. There is a very edge case where it can be a problem as you mentioned: If another JDK version is downloaded from Android Studio and the default download path is changed to `/Applications/Android Studio.app/Contents/X` where X is an empty folder, then the contents of X will be "`Contents/Home/..`", and basically this will end up with the problem you mentioned. 
Btw, did it work on your machine @jonalmeida with the path that included double asterisk?

Reply:
> Is there another situation we should consider? Any preference towards * or **?

You could dynamically determine the path, e.g.
```sh
# TODO: fix me since this actually returns
# /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
# so you'd have to truncate the result more
export JAVA_HOME="$(find /Applications/Android\ Studio.app -name "java" | grep 'Contents/Home')"
```

But that's also fragile and might fail if there were two JDKs. 🤷 

We could also not suggest this as a reasonable way to set up a JDK. 😉 

> However, with this path, I was not able to make java executable working on my local machine. 

@kycn Two things I can think of that might cause this to not work if it works for Jonathan:
- are you using zsh? This seems like a thing zsh might add over bash
- did you use single quotes or double quotes for `JAVA_HOME="`? If single quotes, I wonder if it'd prevent the shell expansion.

Reply:
@mcomella My shell is zsh and I was using double quotes. 
PATH is stated in bash manual as:
`A colon-separated list of directories in which the shell looks for commands. A zero-length (null) directory name in the value of PATH indicates the current directory. A null directory name may appear as two adjacent colons, or as an initial or trailing colon.`.
I understood this statement asit looks for the directories matching the name, i.e. it looks for `*` as a directory name there. Am I missing something?

Reply:
> We could also not suggest this as a reasonable way to set up a JDK. 😉

Yes! I'm inclined to agreeing with you since it looks like from Kaya's experience, this isn't a good shell feature as I expected it to be. 😄

Happy to close this is in that case.